js-miniapp-sample/src/components/ToolBar.js

```diff
@@ -0,0 +1,108 @@
+//@flow
+import React, { Fragment, useState, useEffect } from "react";
+
+import AppBar from "@material-ui/core/AppBar";
+import IconButton from "@material-ui/core/IconButton";
+import { makeStyles } from "@material-ui/core/styles";
+import Toolbar from "@material-ui/core/Toolbar";
+import Typography from "@material-ui/core/Typography";
+import CloseIcon from "@material-ui/icons/Close";
+import MenuIcon from "@material-ui/icons/Menu";
+import { connect } from "react-redux";
+import { withRouter } from "react-router-dom";
+
+import Drawer from "./Drawer";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    flexGrow: 1,
+  },
+  menuButton: {
+    marginRight: theme.spacing(2),
+  },
+  title: {
+    flexGrow: 1,
+  },
+}));
+
+type ToolBarProps = {
+  title: string,
+  showDrawer: Boolean,
+  actions: any,
+  onShrinkToggle: Function,
+  onDrawerToggle: Function,
+  navItems: any,
+};
+
+const ToolBar = (props: ToolBarProps) => {
+  const classes = useStyles();
+  const [drawer, setDrawer] = useState({ show: props.showDrawer ?? false });
```

Comment:
If `drawer` is an object that just has one boolean field, we could probably just use a boolean value instead for simplification:

```suggestion
  const [showDrawer, setShowDrawer] = useState(props.showDrawer ?? false);
```